CvGameCoreDLL_Expansion2/CvUnit.cpp

```diff
@@ -3425,16 +3425,32 @@ bool CvUnit::isActionRecommended(int iAction)
 		CvAssert(eBuild != NO_BUILD);
 		CvAssertMsg(eBuild < GC.getNumBuildInfos(), "Invalid Build");
 
-		//fake this, we're really only interested in one plot
-		ReachablePlots plots;
-		plots.insertWithIndex(SMovePlot(plot()->GetPlotIndex()));
-		map<int, ReachablePlots> allplots;
-		allplots[this->GetID()] = plots;
+		vector<BuilderDirective> directives = GET_PLAYER(getOwner()).GetBuilderTaskingAI()->GetDirectives();
 
-		BuilderDirective aDirective = GET_PLAYER(getOwner()).GetBuilderTaskingAI()->EvaluateBuilder(this,allplots);
-		if(aDirective.m_eDirective != BuilderDirective::NUM_DIRECTIVES && aDirective.m_eBuild == eBuild)
+		if (directives.empty())
+			return false;
+
+		for (vector<BuilderDirective>::iterator it = directives.begin(); it != directives.end(); ++it)
 		{
-			return true;
+			BuilderDirective eDirective = *it;
+			CvPlot* pDirectivePlot = GC.getMap().plot(eDirective.m_sX, eDirective.m_sY);
+
+			if (plot() != pDirectivePlot)
+				continue;
+
+			bool bCanBuild = GET_PLAYER(getOwner()).GetBuilderTaskingAI()->EvaluateBuilder(this, eDirective);
+
+			if (!bCanBuild)
+				continue;
+
+			if (eDirective.m_eBuild == eBuild)
```

Comment:
Is there any reason? Why not? `return eDirective.m_eBuild == eBuild`

Reply:
Good catch! Previously there was only the "return true" statement :)

Reply:
Will include simplification is next commit, but waiting to see if there are any minor things that turn up first.